clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1024,12 +1024,14 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
             FPM.addPass(BoundsCheckingPass());
           });
 
-    if (LangOpts.Sanitize.has(SanitizerKind::Realtime))
+    if (LangOpts.Sanitize.has(SanitizerKind::Realtime)) {
```

Comment:
@cjappl 
@why do we use this location and not `addSanitizers`?  In particular registerOptimizerLastEPCallback?

Reply:
I'll investigate if moving it is possible! Off the top of my head I don't know if there is a reason.

I'll report back with PRs on either/both of these soon.

Reply:
You need to consider performance vs quality.

For other sanitizers it's essentially:
1. If it's early - better precision - in sense we detect issues in C/C++ program which can be optimized out. (can Realtime related issue be optimized out, and do you care about them at all)
2. If it's late - better performance, but we may miss bugs
I believe performance is ~2x difference, so we can't afford that e.g. in Asan.